### PR TITLE
Clean commit for review.

### PR DESCRIPTION
**Pull request by caseyflynn-google**
Creating new PR so only editor tab additions / changes are visible.

### CODE REVIEW

File: bundles/org.eclipse.orion.client.ui/web/css/layout_common.css [hunk 1]
```diff
@@ -354,14 +354,60 @@ hard-coded numbers elsewhere.
 	height: 100%;
 }
 
-.editorViewerHeader {
-	display:flex;
-	background: #3B4B54;
+ .editorViewerHeader {
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/css/layout_common.css]: Do these new style declarations work with the container themeing?

File: bundles/org.eclipse.orion.client.ui/web/css/layout_common.css [hunk 2]
```diff
@@ -354,14 +354,60 @@ hard-coded numbers elsewhere.
...
+
+.editorTabContainer {
+	align-items: flex-start;
+	display: flex;
+	flex: 1;
+	flex-direction: row;
+	list-style: none;
+	margin: 0;
+	outline: none;
+	overflow-x: scroll;
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/css/layout_common.css]: Should we make it overflow-x: hidden to disable the scroll wheel for now?

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js [hunk 3]
```diff
@@ -213,24 +216,327 @@ objects.mixin(TextModelPool.prototype, {
...
+					}.bind(this));
+				}
+			}
+		}.bind(this);
+		
+		this.breadcrumbUniquifier++;
+		var breadcrumbOptions = {
+			container: localBreadcrumbNode,
+			resource: metadata,
+			workspaceRootSegmentName: "Orion Content",
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: The "Orion Content" looks very odd in teh tooltip path names.  How about either 'workspace_root' or just the empty string?  I'd argue the empty string would be better.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js [hunk 4]
```diff
@@ -241,79 +547,22 @@ function EditorViewer(options) {
...
-	this.localBreadcrumbNode = document.createElement("div"); //$NON-NLS-0$
-	var tipContainer = this.fileNodeTooltip.contentContainer();
-	tipContainer.appendChild(this.localBreadcrumbNode);
-	
 	domNode.appendChild(headerNode);
 	
+	this.tabWidget = new TabWidget({parent: headerNode, commandRegistry: this.commandRegistry, openEditor: this.activateContext.openEditor.bind(this.activateContext), id: this.id, fileClient: this.fileClient});
+	
+	this.tabWidget.addEventListener("TabClosed", function(evt) {
+		//TODO(caseyflynn): This is a bug if the file is in use in more than one editor.
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: What happens if its in use in more than one editor?  Should we close it in both editors for now?

File: bundles/org.eclipse.orion.client.ui/web/css/layout_common.css (hunk 1 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/css/layout_common.css]: Yes, I added entries to LightPage.js OrionPage.js and ScopeList.js

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js (hunk 4 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: If used in more than one editor, the undo stack and text model are reloaded. I should have fixed this bug before submitting this PR, will update.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js (hunk 3 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: I agree that "Orion Content" looks odd, but I did this to match the current breadcrumb. I have updated to actually look up the project name as opposed to using a static value. If you like I can remove this entry entirely. Let me know.

File: bundles/org.eclipse.orion.client.ui/web/css/layout_common.css (hunk 2 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/css/layout_common.css]: Changed to overflow-x: auto, this will now hide when not in use.

Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/css/layout_common.css]: Disabled per our conversation.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js (hunk 4 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Bug fixed.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js [hunk 5]
```diff
@@ -205,32 +208,345 @@ objects.mixin(TextModelPool.prototype, {
...
+						}
+					}.bind(this));
+				}
+			}
+		}.bind(this);
+		
+		var breadcrumbOptions = {
+			container: localBreadcrumbNode,
+			resource: metadata,
+			workspaceRootSegmentName: this.fileClient.fileServiceName(metadata.Location),
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: What would the tab header be here?  I really think it should just be $FILEPATH. So at root its just $file_name (with no prefix) or a directory then path/to/directory/file_name.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js [hunk 6]
```diff
@@ -205,32 +208,345 @@ objects.mixin(TextModelPool.prototype, {
...
+	objects.mixin(this, options);
+	this.selectedFile = null;
+	this.commandRegistry = options.commandRegistry;
+	
+	this.fileList = [];
+	this.editorTabs = {};
+	this.breadcrumbUniquifier = "_editor_" + this.id;
+	
+	this.editorTabsEnabled = true;
+	this.maxEditors = 0;
```
Comment (kangy-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Is this variable used anywhere?

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js [hunk 7]
```diff
@@ -205,32 +208,345 @@ objects.mixin(TextModelPool.prototype, {
...
+		editorTab.addEventListener("click", function() {
+			window.location = href;
+ 		});
+ 		
+		editorTab.addEventListener("dragstart", function(e) {
+			this.beingDragged = e.target;
+			setTimeout(function() {
+				tipContainer.classList.add('hideOnDrag');
+				e.target.classList.add('hideOnDrag');
+			});
```
Comment (kangy-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Why do these need to be in a setTimeout?

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js [hunk 8]
```diff
@@ -205,32 +208,345 @@ objects.mixin(TextModelPool.prototype, {
...
+		}.bind(this));
+		
+		editorTab.addEventListener("dragend", function(e) {
+			this.beingDragged = null;
+			tipContainer.classList.remove('hideOnDrag');
+			e.srcElement.classList.remove('hideOnDrag');
```
Comment (kangy-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Is there a reason to not use e.target to obtain the object? srcElement looks non-standard.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js [hunk 9]
```diff
@@ -205,32 +208,345 @@ objects.mixin(TextModelPool.prototype, {
...
+		}.bind(this));
+		
+		editorTab.addEventListener("dragenter", function(e) {
+			e.preventDefault();
+			var target = e.target;
+			var draggedNode = this.beingDragged;
+			if (target.hasAttribute("draggable")) {
```
Comment (kangy-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Should we check the classname of the target? draggable object may not necessarily be a tab.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js (hunk 5 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Acknowledged.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js (hunk 6 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Used in the upcoming PR (will add shortly).

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js (hunk 7 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: If we "hide" them (that is, transform them offscreen) before the drag starts then the drag event ends immediately. By executing this on the next tick it allows the drag event to begin before moving the source elements off screen.

File: bundles/org.eclipse.orion.client.ui/web/edit/setup.js (hunk 9 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/edit/setup.js]: Done.

**Comment by caseyflynn-google** [comment]
Closing PR

**Pull request closed without merging**